Sources/Datadog/URLSessionAutoInstrumentation/Interception/URLFiltering/FirstPartyURLsFilter.swift

```diff
@@ -13,9 +13,11 @@ internal struct FirstPartyURLsFilter {
     private let regex: String
 
     init(hosts: Set<String>) {
```

Comment:
If somehow the Set passed in here stores an order based on how it was build, especially as we build a Set from an Array in the related tests, and if you were able to spot the bug by playing with the input order in the test case. Should we also add a test case with the same input in a different order?

Reply:
Not sure if I get it right, but as this is a `Set`, there should be no difference if running the test against `[h1, h2, h3]` or `[h3, h2, h1]` vectors. Also, providing different permutations of the same vector to initialize `Set` would be very unobvious.

Maybe, to solidify this test, we could run it multiple times? 
```swift
(0..<50).forEach { _ in
   // run the test
}
```
This could be even turned into convention for all tests working with order-agnostic inputs.

Reply:
i too first thought something similar to @acgh 's proposal: passing an array instead of set and test different orders. but keeping it as set makes more sense to me. 
running it multiple times as @ncreated suggests sounds better as a patch (no guarantee for correctness but will probably work), i went that way ✅